packages/web3/src/BaseContractShim.ts

```diff
@@ -1,63 +1,65 @@
 import { BytesLike, ethers } from 'ethers'
 import { dlogger } from '@towns-protocol/dlog'
-
+import { Connect } from './types/typechain'
 export type PromiseOrValue<T> = T | Promise<T>
 
 export const UNKNOWN_ERROR = 'UNKNOWN_ERROR'
 
 const logger = dlogger('csb:BaseContractShim')
 
-// V2 smart contract shim
-// todo: replace BaseContractShim with this when refactoring is done
-export class BaseContractShim<
-    T_DEV_CONTRACT extends ethers.Contract,
-    T_DEV_INTERFACE extends ethers.utils.Interface,
-> {
+export class BaseContractShim<T_CONTRACT extends ethers.Contract> {
```

Comment:
```suggestion
export class BaseContractShim<
  F = unknown,
  T_CONTRACT extends ethers.Contract = F extends { connect: Connect<unknown> } 
    ? ContractType<typeof F.connect> 
    : ethers.Contract
> {
```
Consider this for simplicity. Use it as
```typescript
export class IDropFacetShim extends BaseContractShim<IDropFacet__factory>
```

Reply:
I'll try it out!